src/org/moeaframework/core/variable/BinaryRealStepVariable.java

```diff
@@ -0,0 +1,242 @@
+     * @param upperBound the upper bound of this decision variable, inclusive
+     * @param gray       if the binary representation uses gray coding
+     */
+    public BinaryRealStepVariable(double lowerBound, double upperBound, double step, boolean gray) {
+        super(getNumberOfBits(lowerBound, upperBound, step));
+        this.lowerBound = lowerBound;
+        this.upperBound = upperBound;
+        this.step = step;
+        this.gray = gray;
+        this.numSteps=(upperBound-lowerBound)/step;
+    }
+
+    /**
+     * Constructs an integer-valued variable in the range
+     * {@code lowerBound <= x <= upperBound} with the specified initial value.
+     * Uses gray coding by default.
+     *
+     * @param value      the initial value of this decision variable
+     * @param lowerBound the lower bound of this decision variable, inclusive
+     * @param upperBound the upper bound of this decision variable, inclusive
+     * @param gray       if the binary representation uses gray coding
+     * @throws IllegalArgumentException if the value is out of bounds
+     *                                  {@code (value < lowerBound) || (value > upperBound)}
+     */
+    public BinaryRealStepVariable(double value, double lowerBound, double upperBound, double step,
+                                  boolean gray) {
+        this(lowerBound, upperBound, step, gray);
+        setValue(value);
+    }
+
+    /**
+     * Returns the minimum number of bits required to represent an integer
+     * within the given bounds.
+     *
+     * @param lowerBound the lower bound
+     * @param upperBound the upper bound
+     * @return the minimum number of bits required to represent an integer
+     * within the given bounds
+     */
+    public static final int getNumberOfBits(double lowerBound, double upperBound, double step) {
+        return Integer.SIZE - Integer.numberOfLeadingZeros((int) ((upperBound - lowerBound) / step));
+    }
+
+    /**
+     * Returns the current value of this decision variable.
+     *
+     * @return the current value of this decision variable
+     */
+    public double getValue() {
+        if (gray) {
+            EncodingUtils.grayToBinary(this);
+        }
+
+        double value = EncodingUtils.decode(this);
+
+        if (gray) {
+            EncodingUtils.binaryToGray(this);
+        }
+
+        // if difference is not a power of 2, then the decoded value may be
+        // larger than the difference
+        if (value > numSteps) {
+            value = value % numSteps;
+        }
+
+        return lowerBound + value*step;
+    }
+
+    /**
+     * Sets the value of this decision variable.
+     *
+     * @param value the new value for this decision variable
+     * @throws IllegalArgumentException if the value is out of bounds
+     *                                  {@code (value < getLowerBound()) || (value > getUpperBound())}
+     */
+    public void setValue(double value) {
+        double value1=value;
```

Comment:
Is there a reason for adjusting the value if it's out of bounds?  Should this throw an exception instead?